ui/src/main/java/com/kotcrab/vis/ui/FocusManager.java

```diff
@@ -37,8 +37,8 @@
 	public static void switchFocus (Stage stage, Focusable widget) {
 		if (focusedWidget == widget) return;
 		if (focusedWidget != null) focusedWidget.focusLost();
-		focusedWidget = widget;
 		if (stage != null) stage.setKeyboardFocus(null);
```

Comment:
I'm thinking about this more... I think we need to set `focusedWidget` to null before calling `setKeyboardFocus` otherwise the `focusedWidget` will have `focusLost` called twice.

Reply:
So like this
```
		if (focusedWidget == widget) return;
		if (focusedWidget != null) focusedWidget.focusLost();
		focusedWidget = null;
		if (stage != null) stage.setKeyboardFocus(null);
		focusedWidget = widget;
		focusedWidget.focusGained();
```

Reply:
Yes, you've right